handlers/create.go

```diff
@@ -31,6 +31,11 @@ func CreateMachine(event *events.Event, apiClient *client.RancherClient) error {
 		return handleByIdError(err, event, apiClient)
 	}
 
+	// If name is not supplied, error out
```

Comment:
Yeah, I think the problem is that each driver could require it's own set of validation for name.  The only thing requires is that name should not be empty.  The fix should be on the cattle side though.  I added this to quickly protect go-machine given that it will always fail here anyways.